src/libs/WorkflowUtils.ts

```diff
@@ -191,22 +196,48 @@ function convertApprovalWorkflowToPolicyEmployees({approvalWorkflow, membersToRe
 
     approvalWorkflow.approvers.forEach((approver, index) => {
         const nextApprover = approvalWorkflow.approvers.at(index + 1);
+        const forwardsTo = type === CONST.APPROVAL_WORKFLOW.TYPE.REMOVE ? '' : nextApprover?.email ?? '';
+
+        if (previousEmployeeList[approver.email]?.forwardsTo === forwardsTo) {
+            return;
+        }
+
         updatedEmployeeList[approver.email] = {
             email: approver.email,
-            forwardsTo: type === CONST.APPROVAL_WORKFLOW.TYPE.REMOVE ? '' : nextApprover?.email ?? '',
+            forwardsTo,
         };
     });
 
     approvalWorkflow.members.forEach(({email}) => {
+        const submitsTo = type === CONST.APPROVAL_WORKFLOW.TYPE.REMOVE ? '' : firstApprover.email ?? '';
+
+        if (previousEmployeeList[email]?.submitsTo === submitsTo) {
+            return;
+        }
+
+        if (updatedEmployeeList[email]) {
+            updatedEmployeeList[email].submitsTo = submitsTo;
+            return;
+        }
+
         updatedEmployeeList[email] = {
-            ...(updatedEmployeeList[email] ? updatedEmployeeList[email] : {email}),
-            submitsTo: type === CONST.APPROVAL_WORKFLOW.TYPE.REMOVE ? '' : firstApprover.email ?? '',
+            email,
+            submitsTo,
         };
     });
 
     membersToRemove?.forEach(({email}) => {
+        if (previousEmployeeList[email]?.submitsTo === undefined || previousEmployeeList[email]?.submitsTo === '') {
+            return;
+        }
+
+        if (updatedEmployeeList[email]) {
+            updatedEmployeeList[email].submitsTo = '';
+            return;
+        }
+
         updatedEmployeeList[email] = {
-            ...(updatedEmployeeList[email] ? updatedEmployeeList[email] : {email}),
+            email,
             submitsTo: '',
```

Comment:
Should we reset submiTo to default approver instead of empty string

Reply:
[Same as above](https://github.com/Expensify/App/pull/48411#discussion_r1741862980)